lib/kontena/actors/haproxy_spawner.rb

```diff
@@ -0,0 +1,56 @@
+
+module Kontena::Actors
+  class HaproxySpawner < Concurrent::Actor::RestartingContext
+    include Kontena::Logging
+
+    # @param [String] haproxy_bin
+    # @param [String] config_file
+    def initialize(haproxy_bin = '/usr/local/sbin/haproxy', config_file = '/etc/haproxy/haproxy.cfg')
+      @current_pid = nil
+      @haproxy_cmd = [haproxy_bin, '-f', config_file, '-db']
+      @validate_cmd = [haproxy_bin, '-c -f', config_file]
+    end
+
+    # @param [Kontena::Actors::Message] msg
+    def on_message(msg)
+      case msg.action
+      when :update
+        update_haproxy
+      else
+        pass
+      end
+    end
+
+    def update_haproxy
+      if validate_config
+        if current_pid
+          reload_haproxy
+        else
+          start_haproxy
+        end
+      end
```

Comment:
Should there be some `else` here, to report errors on invalid configs? Or those get passed through from the `system` call `stderr`?